### ISSUE TITLE
IPTV stream uses a redirect which is unsuported by jellyfin

### ISSUE THREAD

**Issue by pauloconstanca** [jellyfin/jellyfin#3692]
**Describe the bug**
When trying to play"RTP" channel from this IPTV list ( https://ptiptv.tk/lista.m3u8 ) (Note: This list is 100% legal, from Portugal), Jellyfin will be stuck loading forever. 

After some digging it seems that the stream URL is redirected which is unexpected by jellyfin causing the issue.

**System (please complete the following information):**
 - OS: Ubuntu 18.04 LTS
 - Virtualization: Docker
 - Clients: Browser
 - Browser: Firefox
 - Jellyfin Version: 10.6.0
 - Playback: IPTV Tunner
 - Reverse Proxy: NGINX
 - Base URL: default
 - Networking: host
 - Storage: Local

**To Reproduce**
1. Go to Live TV and add a M3U Tunner with the following URL: https://ptiptv.tk/lista.m3u8
2. Go to the channels list and try to open a channel called "RTP" (in the list "5 RTP")
3. The channel will be stuck loading forever

**Expected behavior**
Channel should load properly

**Logs**
First:
```
[2020-07-24 23:37:14.028 +01:00] [WRN] [28] Emby.Server.Implementations.HttpServer.HttpListenerHost: HTTP Response 200 to "188.251.162.8". Time (slow): 0:00:00.5173666. "http://localhost:8096/jellyfin/LiveTv/Recordings?UserId=80bc5bc4b2834eb4afa521d3dc4db62e&IsInProgress=true&Fields=CanDelete%2CPrimaryImageAspectRatio&EnableTotalRecordCount=false&EnableImageTypes=Primary%2CThumb%2CBackdrop"
[2020-07-24 23:37:14.175 +01:00] [WRN] [29] Emby.Server.Implementations.HttpServer.HttpListenerHost: HTTP Response 200 to "188.251.162.8". Time (slow): 0:00:00.6655186. "http://localhost:8096/jellyfin/System/Info"
[2020-07-24 23:37:16.663 +01:00] [ERR] [32] Emby.Server.Implementations.HttpServer.HttpListenerHost: Could not find handler for "/jellyfin/socket"
[2020-07-24 23:37:16.664 +01:00] [ERR] [32] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request: "Unable to find the specified file". URL: "http://localhost:8096/jellyfin/socket?deviceId=TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6NzUuMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC83NS4wfDE1ODc0ODcwNzQ2MDQ1"
[2020-07-24 23:37:18.554 +01:00] [ERR] [12] Emby.Server.Implementations.HttpServer.HttpListenerHost: Could not find handler for "/jellyfin/socket"
[2020-07-24 23:37:18.554 +01:00] [ERR] [12] Emby.Server.Implementations.HttpServer.HttpListenerHost: Error processing request: "Unable to find the specified file". URL: "http://localhost:8096/jellyfin/socket?deviceId=TW96aWxsYS81LjAgKFdpbmRvd3MgTlQgMTAuMDsgV2luNjQ7IHg2NDsgcnY6NzUuMCkgR2Vja28vMjAxMDAxMDEgRmlyZWZveC83NS4wfDE1ODc0ODcwNzQ2MDQ1"
```

More Details:
```
[2020-07-24 23:41:37.497 +01:00] [INF] [33] Emby.Server.Implementations.LiveTv.TunerHosts.M3UTunerHost: Opening SharedHttpStream Live stream from "https://ptiptv.tk/rtp1.m3u8|user-agent=Mozilla/5.0"
[2020-07-24 23:41:37.695 +01:00] [ERR] [33] Emby.Server.Implementations.LiveTv.TunerHosts.M3UTunerHost: Error opening tuner
MediaBrowser.Model.Net.HttpException: Temporary Redirect
   at Emby.Server.Implementations.HttpClientManager.HttpClientManager.EnsureSuccessStatusCode(HttpResponseMessage response, HttpRequestOptions options)
   at Emby.Server.Implementations.HttpClientManager.HttpClientManager.SendAsyncInternal(HttpRequestOptions options, HttpMethod httpMethod)
   at Emby.Server.Implementations.HttpClientManager.HttpClientManager.SendAsync(HttpRequestOptions options, HttpMethod httpMethod)
   at Emby.Server.Implementations.LiveTv.TunerHosts.SharedHttpStream.Open(CancellationToken openCancellationToken)
   at Emby.Server.Implementations.LiveTv.TunerHosts.BaseTunerHost.GetChannelStream(String channelId, String streamId, List1 currentLiveStreams, CancellationToken cancellationToken)
```

**Comment by pauloconstanca** [comment]
Hello. Just to check in, this issue still happens on the latest version (10.6.4) of jellyfin.

**Comment by crobibero** [comment]
I just tested this with latest master- I don't see any channel "5 RTP", but "RTP 3" plays back fine for me. I also unsuccessfully attempted to play back "RTP 1", but even VLC wouldn't play that for me.

**Comment by pauloconstanca** [comment]
> I just tested this with latest master- I don't see any channel "5 RTP", but "RTP 3" plays back fine for me. I also unsuccessfully attempted to play back "RTP 1", but even VLC wouldn't play that for me.

Hello crobibero.

I just checked, RTP 3 is not working for me on Jellyfin, as well as RTP. Aparently after checking, this channel (RTP) has a geo-location restriction, it will only load for Portugal. Just tested on my VLC and both of them load up fine, but on jellyfin it's stuck forever "loading" due to the weird redirect it uses.

Also it seems RTP3 has a different error from RTP: Jellyfin will try to load and spit out the following error:

```
Playback Error
This client isn't compatible with the media and the server isn't sending a compatible media format.
```

Hope this helps.

**Comment by crobibero** [comment]
There has been a lot of work on the server from 10.6 to 10.7 (current master / unstable). I only tested 10.7, and it looks like this is already fixed.

**Comment by pauloconstanca** [comment]
Hello everyone. Just confirmed, this has been fixed in 10.7.0. Good work everyone! :)
